### DISCUSSION TITLE
Support for helm lookup with google storage (helm-gcs)

### DISCUSSION THREAD

**Discussion by leevs** [renovatebot/renovate#24915] [Suggest an Idea]
### What would you like Renovate to be able to do?

Helm has a plugin called [helm-gcs](https://github.com/hayorov/helm-gcs) this plugin allows you to manage a private helm repo on Google Cloud Storage. Making it a very cost-effective way. 

At this moment it's unclear how this implementation would work (either it would already work and I was not lucky yet with the config) but currently I am doing the following:

`config.js`
```
hostRules: [
        {
        matchHost: 'gs',
        username: '_json_key_base64',
        password: '<base 64 encoded service token json>`
       }],
```

Also tried to first add the repo's on the system by doing, however I believe renovate would ignore this.
```shell
helm repo add repo-name gs://bucket/path
helm repo update
```

Currently it looks at the `Chart.yaml` and find the dependant charts, but it can't do the lookup of version of those helm repo's. 

Logs:
```
DEBUG: Found package lookup warnings (repository=username/repo, branch=renovate/xxx)
       "warnings": 
         "Failed to look up helm dependency <dep 1 chart>",
         "Failed to look up helm dependency <dep 2 chart>",
       ],
       "warningFiles": ["base/Chart.yaml"]
```

### If you have any ideas on how this should be implemented, please tell us here.

Possible solutions:
- In case this would already work: documentation on how to use it together with `helm-gcs`
- Use local configured helm repo's

### Is this a feature you are interested in implementing yourself?

No

**Comment by leevs** [comment]
#19927
After a discussion on [Renovate discussion 19927](https://github.com/renovatebot/renovate/discussions/19927) , it seems to be related to the protocol being used. 
Changing `gs://` to `https://storage.googleapis.com/` works.  

So not sure how this should be implemented (either by documentation that `gs://` is not supported) or where the validation does allow a different protocol like `gs://`

**Comment by viceice** [comment]
We can probably simply messaging that on extract, so the datasource sees the `https` url instead
